Fix Pixelfed interactions (Likes)

pfefferle opened the pull request:
This PR fixes handling of ActivityPub activities from Pixelfed and other platforms by treating activities with no recipients as public, which aligns with the ActivityPub specification.

Fixes https://wordpress.org/support/topic/boosts-reposts-not-working-in-pixelfed/

## Proposed changes:
- Updated `is_activity_public()` function to return `true` when an activity has no recipients
- Added test coverage for Like activities with trailing slashes in object URLs (common with Pixelfed)
- Updated existing tests to reflect the new behavior where activities without recipients are treated as public

### Other information:

- [X] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [X] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [X] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Fixed compatibility with Pixelfed and similar platforms by treating activities without recipients as public, ensuring boosts and reposts work correctly.

</details>

pfefferle committed:
Normalize trailing slashes in Like activity URLs

Ensures object URLs in Like activities are normalized by removing trailing slashes before processing, improving compatibility with platforms like Pixelfed. Adds tests to verify correct handling and storage of Like activities with trailing slashes in object URLs.

pfefferle committed:
Treat activities with no recipients as public

Added a check in is_activity_public to return true when the recipients list is empty, ensuring activities without specified recipients are considered public.

pfefferle committed:
Treat activities with no recipients as public

Updated logic to treat activities without recipients as public, delivering them to all local actors. Adjusted related tests to reflect this behavior and removed unnecessary untrailingslashit calls for URL normalization.

matticbot committed:
Add changelog

obenland commented on includes/functions.php:
```diff
@@ -607,6 +607,10 @@ function is_activity_public( $data ) {
 
 	$recipients = extract_recipients_from_activity( $data );
 
+	if ( empty( $recipients ) ) {
+		return true;
+	}
```
@pfefferle This feels like valid feedback. Is there a reason `get_activity_visibility()` should not account for empty recipients?

obenland commented on tests/phpunit/tests/includes/collection/class-test-inbox.php:
```diff
@@ -76,8 +76,9 @@ public function test_add_activity_with_post_meta() {
 		$this->assertEquals( 'https://remote.example.com/users/testuser', $remote_actor_meta );
 
 		// Test activitypub_content_visibility meta.
+		// Activities with no recipients are treated as public.
```
```suggestion
		// Activities with no recipients are treated as public.
```

pfefferle committed:
Update tests/phpunit/tests/includes/collection/class-test-inbox.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Treat activities with no recipients as public

Updated get_activity_visibility to return public visibility for activities without recipients. Adjusted related unit test to reflect this behavior change, ensuring empty activities are now considered public.
